src/containers/NoteForm/NoteForm.js

```diff
@@ -3,22 +3,33 @@ import { connect } from 'react-redux'
 import { postNote } from '../../thunks/postNote'
 import { Redirect, Link } from 'react-router-dom'
 import { NoteItem } from '../NoteItem/NoteItem'
+import { editNote } from '../../thunks/editNote'
+import { deleteNote } from '../../thunks/deleteNote'
 
 export class NoteForm extends Component {
-  constructor() {
-    super()
+  constructor(props) {
+    super(props)
     this.state = {
       note: {
-        title: '',
-        id: Date.now()
+        title: this.props.note.title,
+        id: this.props.note.id
       },
-      items: [],
+      items: this.props.items,
+      isDeleted: false,
       redirect: false
     }
   }
 
   componentDidMount = () => {
-    this.setState({ items: [{ id: Date.now(), description: '', noteID: this.state.note.id }] })
+    !this.props.isEdit && this.setState({
+      items: [{ id: Date.now(), description: '', noteID: this.state.note.id }]
+    })
```

Comment:
I like the differentiation between when to set up the extra item field.
After thinking about it I think it may be worthwhile to figure out how to get rid of this and only add a field if the title/item above is typed in